src/shared/util.js

```diff
@@ -1150,6 +1150,38 @@ function createPromiseCapability() {
   return capability;
 }
 
+/**
+ * Use binary search to find the index of the first item in a given array which
+ * passes a given condition. The items are expected to be sorted in the sense
+ * that if the condition is true for one item in the array, then it is also true
+ * for all following items.
+ *
+ * @returns {number} Index of the first array element to pass the test,
+ *                   or |items.length| if no such element exists.
+ */
+function binarySearchFirstItem(items, condition, start = 0) {
```

Comment:
Are you expecting to use this code anywhere in the `src/core/`-folder?

If not, you probably want to place this in https://github.com/mozilla/pdf.js/blob/master/src/display/display_utils.js instead to avoid unnecessarily duplicating this code in both of the *built* `pdf.js` and `pdf.worker.js` files.